clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2681,8 +2681,11 @@ public void handleResponse(AbstractResponse abstractResponse) {
                     if (descriptions.size() > 0) {
                         future.complete(descriptions);
                     } else {
-                        // descriptions will be empty if and only if the user is not authorized to describe cluster resource.
-                        future.completeExceptionally(Errors.CLUSTER_AUTHORIZATION_FAILED.exception());
+                        // Up to v3 DescribeLogDirsResponse did not have an error code field, hence it defaults to None
+                        Errors error = response.data().errorCode() == Errors.NONE.code()
```

Comment:
Should we extend unit tests in `KafkaAdminTest` to cover this change?

Reply:
Absolutely, I've updated the tests.

Reply:
@mimaison What would happen if DescribeLogDirsResponse v3 actually returns NONE as an error (the default value)? It seems to be that we need to take the version into account here as well.

Reply:
With v3, if `NONE` is returned there should be some logdirs in the response so we should have entered the if condition above.

Reply:
That makes sense, thanks.